### ISSUE TITLE
feat: Add Logistic Regression classifier

### ISSUE THREAD

**Issue by cardmagic** [cardmagic/classifier#102]
## Summary

Add a Logistic Regression (MaxEnt) classifier as a complement to Naive Bayes. Logistic Regression is often the strongest "classic" baseline for text classification, offering better accuracy than Naive Bayes while remaining fast and interpretable.

## Motivation

- **More accurate than Naive Bayes** on many text problems (no independence assumption)
- **Probabilistic outputs** - calibrated class probabilities
- **Interpretable** - feature weights show which words matter for each class
- **Fast** - linear time prediction, efficient training with SGD
- **Well-understood** - the standard baseline in NLP/ML before deep learning

## Proposed API

```ruby
classifier = Classifier::LogisticRegression.new(:spam, :ham)

# Training (same interface as Bayes)
classifier.train(spam: ["Buy now!", "Free money!!!"])
classifier.train(ham: ["Meeting tomorrow", "Project update"])

# Classification
classifier.classify("Claim your prize!")
# => "spam"

# Probabilities (better calibrated than Bayes)
classifier.probabilities("Claim your prize!")
# => {"spam" => 0.92, "ham" => 0.08}

# Feature weights (interpretability)
classifier.weights(:spam)
# => {free: 2.3, buy: 1.8, money: 1.5, ...}
```

## Implementation Notes

- Use stochastic gradient descent (SGD) for training
- L2 regularization to prevent overfitting
- Leverage existing `word_hash` infrastructure for feature extraction
- Consider TF-IDF weighting (see #102 if created)
- Pure Ruby implementation feasible; native extension optional for speed

## References

- [Logistic Regression for Text Classification](https://web.stanford.edu/~jurafsky/slp3/5.pdf) - Stanford NLP book chapter
- [scikit-learn LogisticRegression](https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression)

## Related

- Complements Naive Bayes for users who need higher accuracy
- Can use TF-IDF features (#102) for better performance
